src/components/TimePicker/TimePicker.js

```diff
@@ -87,15 +86,19 @@ function TimePicker({forwardedRef, defaultValue, onSubmit, onInputChange}) {
     const [selectionHour, setSelectionHour] = useState({start: 0, end: 0});
     const [selectionMinute, setSelectionMinute] = useState({start: 2, end: 2}); // we focus it by default so need  to have selection on the end
     const [hours, setHours] = useState(() => DateUtils.get12HourTimeObjectFromDate(value).hour);
-    const [minute, setMinute] = useState(() => DateUtils.get12HourTimeObjectFromDate(value).minute);
+    const [minutes, setMinutes] = useState(() => DateUtils.get12HourTimeObjectFromDate(value).minute);
     const [amPmValue, setAmPmValue] = useState(() => DateUtils.get12HourTimeObjectFromDate(value).period);
+    const [lastPressedKey, setLastPressedKey] = useState('');
```

Comment:
I don't see the need for this new state. The backspace bug is not in main so there must be a better way to handle it

Reply:
> The backspace bug is not in main

This is incorrect. This new state is used to differentiate between removing left or right character when the caret is placed between 2 digits. You can try it yourself in Prod – the bug exists. With my fix using this state – everything works correctly.

Reply:
But on main we weren't able to insert 3 digits in the hours case

Reply:
I've already fixed this. It's not related to `lastPressedKey`. I just needed to use the `.substring(0,2)` in the last if-else branch.

Reply:
Ah okay let's make lastPressedKey a ref (useRef) instead of a state. I don't think we need to re-render on key press